esphome/components/display/display_buffer.cpp

```diff
@@ -392,6 +402,484 @@ void DisplayBuffer::strftime(int x, int y, Font *font, const char *format, time:
+  //   2) a flat top triangle
+  // - Sort the vertices in descending vertical position.
+  //   This serves two purposes:
+  //   1) ensures that the division between the flat bottom
+  //      and flat top triangles occurs at Y=Y1
+  //   2) ensure that we avoid division-by-zero in the for loops
+  // - Walk each scan line and determine the intersection
+  //   between triangle side A & B (flat bottom triangle)
+  //   and then C and B (flat top triangle) using line slopes.
+
+  // Sort vertices
+  // - Want y0 >= y1 >= y2
+  if (y2 > y1) {
+    this->swap_coords_(&x2, &y2, &x1, &y1);
+  }
+  if (y1 > y0) {
+    this->swap_coords_(&x0, &y0, &x1, &y1);
+  }
+  if (y2 > y1) {
+    this->swap_coords_(&x2, &y2, &x1, &y1);
+  }
+
+  // TODO: It is more efficient to calculate row endpoints
+  // using incremental additions instead of multiplies/divides
+
+  int16_t xa, xb, xc, yos;
+  int16_t x01, x20, y01, y20, x21, y21;
+  x01 = x0 - x1;
+  y01 = y0 - y1;
+  x20 = x2 - x0;
+  y20 = y2 - y0;
+  x21 = x2 - x1;
+  y21 = y2 - y1;
+
+  // Flat bottom scenario
+  // NOTE: Due to vertex sorting and loop range, it shouldn't
+  // be possible to enter loop when y0 == y1 or y2
+  for (yos = 0; yos < y01; yos++) {
+    // Determine row endpoints (no rounding)
+    // xa = (yos          )*(x0-x1)/(y0-y1);
+    // xb = (yos-(y0-y1))*(x2-x0)/(y2-y0);
+
+    // Determine row endpoints (using rounding)
+    xa = 2 * yos * x01;
+    xa += (xa >= 0) ? abs(y01) : -abs(y01);
+    xa /= 2 * y01;
+
+    xb = 2 * (yos - y01) * x20;
+    xb += (xb >= 0) ? abs(y20) : -abs(y20);
+    xb /= 2 * y20;
+
+    // Draw horizontal line between endpoints
+    this->line(x1 + xa, y1 + yos, x0 + xb, y1 + yos, color);
+  }
+
+  // Flat top scenario
+  // NOTE: Due to vertex sorting and loop range, it shouldn't
+  // be possible to enter loop when y2 == y0 or y1
+  for (yos = y21; yos < 0; yos++) {
+    // Determine row endpoints (no rounding)
+    // xc = (yos          )*(x2-x1)/(y2-y1);
+    // xb = (yos-(y0-y1))*(x2-x0)/(y2-y0);
+
+    // Determine row endpoints (using rounding)
+    xc = 2 * yos * x21;
+    xc += (xc >= 0) ? abs(y21) : -abs(y21);
+    xc /= 2 * y21;
+
+    xb = 2 * (yos - y01) * x20;
+    xb += (xb >= 0) ? abs(y20) : -abs(y20);
+    xb /= 2 * y20;
+
+    // Draw horizontal line between endpoints
+    this->line(x1 + xc, y1 + yos, x0 + xb, y1 + yos, color);
+  }
+}
+
+void DisplayBuffer::quad(Point *points, Color color) {
+  int16_t x0, y0, x1, y1;
+
+  x0 = points[0].x;
+  y0 = points[0].y;
+  x1 = points[1].x;
+  y1 = points[1].y;
+  this->line(x0, y0, x1, y1, color);
+
+  x0 = points[1].x;
+  y0 = points[1].y;
+  x1 = points[2].x;
+  y1 = points[2].y;
+  this->line(x0, y0, x1, y1, color);
+
+  x0 = points[2].x;
+  y0 = points[2].y;
+  x1 = points[3].x;
+  y1 = points[3].y;
+  this->line(x0, y0, x1, y1, color);
+
+  x0 = points[3].x;
+  y0 = points[3].y;
+  x1 = points[0].x;
+  y1 = points[0].y;
+  this->line(x0, y0, x1, y1, color);
+}
```

Comment:
Given that you've created a `Point`, I'd suggest creating additional function signatures that take a `Point` instead of an (x, y) pair, then this can be greatly simplified:
```suggestion
void DisplayBuffer::quad(Point *points, Color color) {
  this->line(points[0], points[1], color);
  this->line(points[1], points[2], color);
  this->line(points[2], points[3], color);
  this->line(points[3], points[0], color);
}
```

Reply:
can you explain more, could you tell how that new function should be look like?

Reply:
You'd create a function something like this:
```cpp
void DisplayBuffer::line(Point &pt1, Point &pt2, Color color,) {
  this->line(pt1.x, pt1.y, pt2.x, pt2.y, color);
}

Reply:
okey, i got you, I will add it when i see more then 5 thumbs-up otherwise i go the other way around and remove the point version in favor of:
```cpp
void qaud(int16_t x0, int16_t y0, int16_t x1, int16_t y1, int16_t x2, int16_t y2, int16_t x3, int16_t y3, Color color = COLOR_ON) { 
  ...
}
```

Reply:
I have removed the point structure from the PR changes,